### PR TITLE
Expanding the Experimental MODsuit Bepis Node with three new modules.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
So, I've had this idea to make a contribution to the Bepis feature with some modsuit stuff. The gimmicky stuff is ok and a good way to even out the better content since it has game of chance design it has (you can find those disks in space anyway so...). However, the Experimental MODsuit node feels very underwhelming right now, compared to how big that feature is.

This PR adds three MOD modules to the Experimental MODsuit node, plus two more:
- Magneto Charger: While the Modsuit is activated, each step the user takes will charge the installed power cell by a tiny bit, enough to sustain a standard modsuit of generic slow speed with only a few, easy modules installed. It won't work in zero G, while flying, pulled by someone else, on a conveyor belt, riding a vehicle or crawling on the floor, though.
- Recycler: It collects (most) garbage and casings off the ground and recycles them into material sheets that can be dispensed on an adjacent location or storage with with Middle Mouse Button. Doesn't clean debris, and scuffed because most trash doesn't yield material anyway.
- - It also has two subtypes, unbound from the node: one that dispenses riot foam darts and can be found on the black market, and another that dispenses the more innocuous foam darts, rarely found in maints.
- Shooting Assistant: A configurable module. On Stormtrooper mode, it will give the user a faster fire rate (the double tap trait) at the cost of accuracy. On Sharpshooter mode, it will improve the user accuracy and make their shots ricochet against walls at least once (if the hit atom allows that, that is, e.g. lasers don't ricochet against iron walls), at the cost of movement speed. Both modes also prevent the user from dual wielding guns.

To make the Stormtrooper mode stackable with the poor aim quirk and refrain from making a new trait for the sharpshooter mode, the gun spread code in gun.dm has also received a little refactor and cleanup. Also, it's been tested.

## Why It's Good For The Game
The Experimental MODsuit node is quite shabby and could use something extra to make it more appealing to MODsuit enjoyers.

Also doubles down as a small addition to the black market and maint loot, and code cleanup, since gun code gives off some garbled vibes.

## Changelog

:cl:
add: Expanded the Experimental MODsuit Bepis node with three new modules: Magneto Charger, Recycler and Shooting Assistant.
add: Added a Riot Foam Recycler module to the black market, as well a more innocuous version as maint loot.
/:cl:

### CODE REVIEW

File: code/game/objects/effects/spawners/random/mod.dm [hunk 1]
```diff
@@ -12,7 +12,8 @@
 		/obj/item/mod/module/balloon = 1,
 		/obj/item/mod/module/paper_dispenser = 1,
 		/obj/item/mod/module/hat_stabilizer = 2,
-		/obj/item/mod/module/stamp = 1
+		/obj/item/mod/module/stamp = 1,
+		/obj/item/mod/module/recycler/donk/safe = 1
```
Comment (Fikou) [inline code/game/objects/effects/spawners/random/mod.dm:16]:
```suggestion
		/obj/item/mod/module/recycler/donk/safe = 1,
```

**Comment by atlasle** [comment]
why 'magneto' charger and not 'dynamo' charger?

**Comment by Ghommie** [comment]
The Shooting Assistant module is now a configurable passive module, thanks to Fikou's suggestion on the chat.

**Comment by necromanceranne** [comment]
As a suggestion for the sharpshooter, since it's possible it might not have consistent use cases for every weapon (for instance, only detectives really get much use out of it on most stations, or just traitors with their ballistics), maybe make it prevent bullet deviation into limbs as another bonus?

**Comment by Ghommie** [comment]
Oh, we still have that? Sure then.

**Comment by necromanceranne** [comment]
I don't think I ever managed to bring over any code to prevent bullet deviation yet, but I wouldn't suspect it to be too difficult.

**Comment by Ghommie** [comment]
That was actually easy.

**Comment by EricZilla** [comment]
Cant wait for the NERF Response unit that all has different dart guns and the riot recycler

**Comment by zeroisthebiggay** [comment]
> > why 'magneto' charger and not 'dynamo' charger?
> 
> [Because bike/bottle dynamos are actually magnetos.](https://en.wikipedia.org/wiki/Bottle_dynamo) Don't ask me how it would work for a MODsuit, I just know that much. 😂

i hold no opinions i just thought that this really reminded me of the [joint torsion ratchet](https://cddawiki.chezzo.com/cdda_wiki/index.php/Joint_Torsion_Ratchetl) from cdda because it's literally the exact same concept and i think it sounds better than magneto whositwhatsit

i hear magneto i think magnets not electricity from walking

**Review by Fikou — APPROVED** [review]

**Review by ArcaneMusic — APPROVED** [review]

**Pull request merged**